ledfx/api/virtuals_tools.py

```diff
@@ -102,21 +151,14 @@ async def put(self, virtual_id, request) -> web.Response:
                 )
 
         if tool == "oneshot":
-            color = parse_color(validate_color(data.get("color", "white")))
-            ramp = data.get("ramp", 0)
-            hold = data.get("hold", 0)
-            fade = data.get("fade", 0)
-            brightness = min(1, max(0, data.get("brightness", 1)))
-
-            # if all values are zero, we will now just ensure any current
-            # oneshot are cancelled
-
-            result = virtual.add_oneshot(
-                Flash(color, ramp, hold, fade, brightness)
-            )
+            result = False
+            for oneshot in virtual.oneshots:
+                if isinstance(oneshot, Flash):
+                    oneshot.active = False
```

Comment:
If you set oneshot.active = False, isn't that leaving the instance orphaned in a list, rather than just destroying the list?

Won't that lead to a cumulative cruft in the _oneshots[] list?

Reply:
Feel bad now, I went looking for that and missed it!